projects/RabbitMQ.Client/client/impl/Connection.cs

```diff
@@ -750,7 +741,7 @@ public void MaybeStartHeartbeatTimers()
 
         public void StartMainLoop()
         {
-            _mainLoopTask = Task.Run((Action)MainLoop);
+            _mainLoopTask = Task.Factory.StartNew(MainLoop, TaskCreationOptions.LongRunning);
```

Comment:
This forces the creation of a new thread instead of blocking one of the threadpool